src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import {
+  CSSTransition,
+  TransitionGroup,
+} from 'react-transition-group';
+import './TodoList.scss';
+
+import { TodoItem } from '../TodoItem';
+
+import { TodoRich } from '../../types/TodoRich';
+import { TodoRichEditable } from '../../types/TodoRichEditable';
+
+type Props = {
+  todos: TodoRich[];
+  tempTodo: TodoRich | null;
+  onTodoDelete: (todoId: number) => Promise<void>;
+  onTodoUpdate: (todoId: number, updatedData: TodoRichEditable)
+  => Promise<void>;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  onTodoDelete,
+  onTodoUpdate,
+}) => (
+  <section className="todoapp__main">
+    <TransitionGroup>
+      {todos.map(todo => (
+        <CSSTransition
+          key={todo.id}
+          timeout={300}
+          classNames="item"
+        >
+          <TodoItem
+            key={todo.id}
```

Comment:
I can be mistaken, but you don't have to pass key here, since you passed it inside of the CSSTransition